src/Mesh/Topography.jl

```diff
@@ -0,0 +1,163 @@
+module Topography
+#
+#    # Topography:
+#
+#    This module contains:
+#
+#    -  ReadExternalHeader(header_file_in)
+#    -  ReadExternalTxtCoordinates(body_file_in, TopoBathy_flg, nlon, nlat)
+#
+#    to read extenral grid files. 
+#
+#    ## The current implementation can only read ASCII files from the NOAA database
+#        url{https://www.ngdc.noaa.gov/mgg/topo/}
+#
+#
+import MPI
+
+using Printf
+using DelimitedFiles
+using Dierckx
+
+export ReadExternalHeader, ReadExternalTxtCoordinates
+
+# {{{ READTOPOtxt_header
+"""   
+    ReadExternalHeader(header_file_in)
+  
+    reads the header file containing the details of the topography from 
+    - NOAA ASCII file
+    - (NCDF and DEM files to be added. See reader in MMesh3D)
+  
+    
+    returns nlon, nlat, lonmin, lonmax, latmin, latmax, dlon, dlat
```

Comment:
```suggestion
returns
 - `nlon` nlon description
 - `nlat` nlat description
 - `lonmin` lonmin description
 - `lonmax` lonmax description
 - `latmin` latmin description
 - `latmax` latmax description
 - `dlon` dlon description
 - `dlat` dlat description
```

Reply:
@smarras79 could you please confirm what these vars are, exactly?

Reply:
Notice: the NOAA topography files are given in lat-lon-height coordinates even if we are currently reading them as if they were just Cartesian x-y-z coordinates:

- `nlon` number of nodal points along the longitude direction 
 - `nlat` number of nodal points along the latitude direction 
 - `lonmin` westernmost longitude coordinate
 - `lonmax` easternmost longitude coordinate
 - `latmin` southernmost latitude coordinate
 - `latmax` northernmost latitude coordinate
 - `dlon` resolution in the longitude direction
 - `dlat` resolution in the latitude direction